src/persistence/db/upgrades/dbto11.cpp

```diff
@@ -0,0 +1,290 @@
+/*
+    Copyright © 2022 by The qTox Project Contributors
+
+    This file is part of qTox, a Qt-based graphical interface for Tox.
+
+    qTox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    qTox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with qTox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "dbto11.h"
+
+#include "src/core/toxpk.h"
+
+#include <QDebug>
+
+bool DbTo11::dbSchema10to11(RawDatabase& db)
+{
+    // splitting peers table into separate chat and authors table.
+    // peers had a dual-meaning of each friend having their own chat, but also each peer
+    // being authors of messages. This wasn't true for self which was in the peers table
+    // but had no related chat. For upcomming group history, the confusion is amplified
+    // since each group is a chat so is in peers, but authors no messages, and each group
+    // member is an author so is in peers, but has no chat.
+    // Splitting peers makes the relation much clearer and the tables have a single meaning.
+    QVector<RawDatabase::Query> upgradeQueries;
+    PeersToAuthors::peersToAuthors(db, upgradeQueries);
```

Comment:
In general there are a lot of missed bool checks in this path. I think we should fail at the top level if we fail to read any of the expected fields

Reply:
Doh, piped the bools up but then never checked them. Fixed.